TestPreemptWithPermitPlugin flakes

<!-- Please only use this template for submitting reports about failing tests in Kubernetes CI jobs -->

**Which jobs are failing**:

pull-kubernetes-integration

**Which test(s) are failing**:

TestPreemptWithPermitPlugin

**Since when has it been failing**:

13:30 PST, Sep 9, 2019

**Testgrid link**:

- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/81229/pull-kubernetes-integration/1159912162789953536/
- https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/81068/pull-kubernetes-integration/1159945886181101569/

**Reason for failure**:

Need investigations.

**Anything else we need to know**:

/sig scheduling

Another flaky [here](https://github.com/kubernetes/kubernetes/pull/81006#pullrequestreview-271138457) https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/81006/pull-kubernetes-integration/1158614573872320512/ 

```go
--- FAIL: TestPreemptWithPermitPlugin (64.95s)
    framework_test.go:1618: Expected the preemptor pod to be scheduled. error: timed out waiting for the condition
    framework_test.go:1622: Expected the waiting pod to get preempted and deleted
```

--- a/test/integration/scheduler/framework_test.go
+++ b/test/integration/scheduler/framework_test.go
@@ -419,8 +419,6 @@ func NewUnreservePlugin(_ *runtime.Unknown, _ framework.FrameworkHandle) (framew
 	return unresPlugin, nil
 }
 
-var perPlugin = &PermitPlugin{}
-
 // Name returns name of the plugin.
 func (pp *PermitPlugin) Name() string {
 	return permitPluginName
@@ -476,10 +474,12 @@ func (pp *PermitPlugin) reset() {
 	pp.allowPermit = false
 }
 
-// NewPermitPlugin is the factory for permit plugin.
-func NewPermitPlugin(_ *runtime.Unknown, fh framework.FrameworkHandle) (framework.Plugin, error) {
-	perPlugin.fh = fh
-	return perPlugin, nil
+// NewPermitPlugin returns a factory for permit plugin with specified PermitPlugin.
+func NewPermitPlugin(permitPlugin *PermitPlugin) framework.PluginFactory {
+	return func(_ *runtime.Unknown, fh framework.FrameworkHandle) (framework.Plugin, error) {
+		permitPlugin.fh = fh
+		return permitPlugin, nil
+	}
 }
 
 // TestPrefilterPlugin tests invocation of prefilter plugins.
@@ -1221,7 +1221,8 @@ func TestPostbindPlugin(t *testing.T) {
 // TestPermitPlugin tests invocation of permit plugins.
 func TestPermitPlugin(t *testing.T) {
 	// Create a plugin registry for testing. Register only a permit plugin.
-	registry := framework.Registry{permitPluginName: NewPermitPlugin}
+	perPlugin := &PermitPlugin{}
+	registry := framework.Registry{permitPluginName: NewPermitPlugin(perPlugin)}
 
 	// Setup initial permit plugin for testing.
 	plugins := &schedulerconfig.Plugins{
@@ -1332,7 +1333,8 @@ func TestPermitPlugin(t *testing.T) {
 // TestCoSchedulingWithPermitPlugin tests invocation of permit plugins.
 func TestCoSchedulingWithPermitPlugin(t *testing.T) {
 	// Create a plugin registry for testing. Register only a permit plugin.
-	registry := framework.Registry{permitPluginName: NewPermitPlugin}
+	perPlugin := &PermitPlugin{}
+	registry := framework.Registry{permitPluginName: NewPermitPlugin(perPlugin)}
 
 	// Setup initial permit plugin for testing.
 	plugins := &schedulerconfig.Plugins{
@@ -1544,7 +1546,8 @@ func TestPostFilterPlugin(t *testing.T) {
 // TestPreemptWithPermitPlugin tests preempt with permit plugins.
 func TestPreemptWithPermitPlugin(t *testing.T) {
 	// Create a plugin registry for testing. Register only a permit plugin.
-	registry := framework.Registry{permitPluginName: NewPermitPlugin}
+	perPlugin := &PermitPlugin{}
+	registry := framework.Registry{permitPluginName: NewPermitPlugin(perPlugin)}
 
 	// Setup initial permit plugin for testing.
 	plugins := &schedulerconfig.Plugins{

